### PR TITLE
Flesh Out Support for ML Model Exec Configs

### PR DESCRIPTION
**Pull request by noanflaherty**
Demo: https://www.loom.com/share/7f6439dfebcb42dd9e20d1041af2cf21

### CODE REVIEW

File: internal/provider/ml_model/model.go [hunk 1]
```diff
@@ -15,14 +16,28 @@ func NewVellumMLModelCreateRequest(ctx context.Context, mlModelModel *TfMLModelR
...
-	features := []vellum.MlModelFeature{"CHAT_MESSAGE_SYSTEM"}
+	features := []vellum.MlModelFeature{}
+	for _, feature := range mlModelModel.ExecConfig.Features.Elements() {
+		feature, _ := vellum.NewMlModelFeatureFromString(feature.(types.String).ValueString())
+		features = append(features, feature)
+	}
+
+	metadata := map[string]interface{}{}
+	for key, tfvalue := range mlModelModel.ExecConfig.Metadata.Elements() {
+		value := tfvalue.(types.String).ValueString()
```
Comment (noanflaherty) [inline internal/provider/ml_model/model.go:27]: For now I'm enforcing that all metadata is a mapping between string <> string

File: internal/sdk/ml_models.go [hunk 2]
```diff
@@ -384,8 +384,8 @@ func (m *MlModelDisplayTagEnumValueLabel) String() string {
 type MlModelExecConfig struct {
 	ModelIdentifier string                  `json:"model_identifier"`
 	BaseUrl         string                  `json:"base_url"`
-	Metadata        map[string]interface{}  `json:"metadata,omitempty"`
-	Features        []MlModelFeature        `json:"features,omitempty"`
+	Metadata        map[string]string       `json:"metadata"`
+	Features        []MlModelFeature        `json:"features"`
```
Comment (noanflaherty) [inline internal/sdk/ml_models.go:388]: Note that due to a bug in the fern Go generator, I had to manually update this sdk code to remove the `,omitempty` (see [here](https://vellum-ai.slack.com/archives/C052UMCSFA8/p1723067503190829)).

The changing of `Metadata        map[string]interface{}` to `Metadata        map[string]string` was done in line with my comment above. It's just pretty hard to manage dynamic nested types in Golang/Terraform, so I decided to simplify for now.

File: internal/provider/ml_model/model.go (hunk 1 above)
Comment (dvargasfuertes) [reply internal/provider/ml_model/model.go:27]: [oos] I wouldn't be totally opposed to just enforcing this on Vellum's end too, now that we'll have consumers dependent on this fact

File: internal/provider/ml_model/resource.go [hunk 3]
```diff
@@ -155,6 +165,34 @@ func (r *MLModelResource) Schema(ctx context.Context, req resource.SchemaRequest
...
+						Description:         "The features",
+						MarkdownDescription: "The features",
+						Required:            true,
+						ElementType:         schema.StringAttribute{}.GetType(),
+					}.GetType(),
+					"metadata": schema.MapAttribute{
+						Description: "Arbitrary JSON object",
+						Required:    true,
+						ElementType: types.StringType,
+					}.GetType(),
```
Comment (dvargasfuertes) [inline internal/provider/ml_model/resource.go:193]: I can't believe this worked lol

File: internal/provider/ml_model/resource.go [hunk 4]
```diff
@@ -155,6 +165,34 @@ func (r *MLModelResource) Schema(ctx context.Context, req resource.SchemaRequest
...
+						Required:            true,
+					}.GetType(),
+					"features": schema.ListAttribute{
+						Description:         "The features",
+						MarkdownDescription: "The features",
+						Required:            true,
+						ElementType:         schema.StringAttribute{}.GetType(),
+					}.GetType(),
+					"metadata": schema.MapAttribute{
+						Description: "Arbitrary JSON object",
```
Comment (dvargasfuertes) [inline internal/provider/ml_model/resource.go:190]: [nit] did we update these descriptions?

**Review by dvargasfuertes — APPROVED** [review]

**Pull request merged**